File: src/models/Result.js
import mongoose from 'mongoose';

const ResultSchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true,
    index: true,
  },
  examId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Exam',
    required: true,
  },
  scorePercent: {
    type: Number,
    required: true,
  },
  correctCount: {
    type: Number,
    required: true,
  },
  wrongCount: {
    type: Number,
    required: true,
  },
  skippedCount: {
    type: Number,
    required: true,
  },
  totalCount: {
    type: Number,
    required: true,
  },
  passed: {
    type: Boolean,
    required: true,
  },
  timeTaken: {
    type: Number, // seconds
    default: 0,
  },
  userAnswers: {
    type: Object, // Stores map of questionId -> array of selected option indices
  }
}, { timestamps: true });

ResultSchema.index({ userId: 1, createdAt: -1 });

export default mongoose.models.Result || mongoose.model('Result', ResultSchema);

File: backend-spring/src/main/java/com/nextlevel/api/model/Result.java
package com.nextlevel.api.model;

import java.time.Instant;
import java.util.List;
import java.util.Map;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.CompoundIndex;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.Data;

@Data
@Document(collection = "results")
@CompoundIndex(def = "{'userId': 1, 'createdAt': -1}")
public class Result {

    @Id
    private String id;
    private String userId;
    private String examId;
    private Double scorePercent;
    private Integer correctCount;
    private Integer wrongCount;
    private Integer skippedCount;
    private Integer totalCount;
    private Boolean passed;
    private Integer timeTaken;
    private Map<String, List<Integer>> userAnswers;
    private Instant createdAt;
}
